service/debugger/locations.go

```diff
@@ -303,8 +304,33 @@ func (loc *NormalLocationSpec) FileMatch(path string) bool {
 	return partialPathMatch(loc.Base, path)
 }
 
-func tryMatchRelativePathByProc(expr, debugname, file string) bool {
-	return len(expr) > 0 && expr[0] == '.' && file == path.Join(path.Dir(debugname), expr)
+func (loc *NormalLocationSpec) fullPathMatch(processArgs []string, file string) bool {
+	locBaseAbsPath := loc.Base
+	if !filepath.IsAbs(locBaseAbsPath) && len(processArgs) >= 1 {
+		locBaseAbsPath = path.Join(path.Dir(processArgs[0]), loc.Base)
+	}
+	if !filepath.IsAbs(locBaseAbsPath) {
+		return false
+	}
+
+	curFile := file
+	for {
+		if curFile == locBaseAbsPath {
+			return true
+		}
+		symLinkFile, err := os.Readlink(curFile)
```

Comment:
I'm concerned with the performance of this. How fast is this going to be for a project that has 100k source files? I'm personally leaning towards not supporting this use case at all.

Reply:
Like, what happens if the file system happens to be a network mountpoint and we stat 100000 files?

Reply:
I test the performance of Readlink. My pc is 4 core i7 cpu / 8G. 
Create 100K files, and then ReadLink all files. [test_source_code](https://gist.github.com/chainhelen/5732fcd52513f835d9f21f857ca4392f).   
The average cost of Readlink for 100k files is approximate 1.5s 
```
1: 1.49343594s
2: 1.502350766s
3: 1.502523769s
4: 1.562780299s
5: 1.599868709s
6: 1.526759598s
7: 1.586828969s
8: 1.527909997s
9: 1.556420267s
10: 1.505442886s
```  

en,  this pr maybe closed,If we consider the performance problem of 100K.

Reply:
I have no good idea for 100k, or just only support realpath and save the realpath when debug is starting.

Reply:
I agree with @aarzilli we could see a huge performance hit with this. I'm also leaning towards not implementing this unless we can find a more performant way.

Reply:
@aarzilli @derekparker   Yes, I agree, thx. 

But I have one more idea, we can add a config to control(default off) and let the user decide(I will finish it tonight).